api/pricers/azure/vm.go

```diff
@@ -52,7 +78,35 @@ func (v *VirtualMachine) GetHourlyPrice(ctx context.Context) float64 {
 	}
 	//Assume that the first one is the one we want
 	if len(vms.Items) > 0 {
-		unitPrice = vms.Items[0].UnitPrice
+		//The unitPrice reflects annual amounts for Reservation instances.  Need to convert this to an hourlyRate
+		if useReservationBilling(*v) {
+			for _, item := range vms.Items {
+				//we can't filter on 'reservationTerm' in the ODATA query, so we need to do it here
+				if item.ReservationTerm == "1 Year" && v.PriceType == Reservation1Yr {
+					unitPrice = item.UnitPrice / common.YEAR_HOURS
+					break
+				} else if item.ReservationTerm == "3 Years" && v.PriceType == Reservation3Yr {
+					unitPrice = item.UnitPrice / (3.0 * common.YEAR_HOURS)
+					break
+				}
+			}
+			if unitPrice == 0.0 {
+				//We couldn't find a match.  Do we need to return this error or simply log it
+				err = errors.New(fmt.Sprintf("Could not find an item with reservation duration %d", v.PriceType))
```

Comment:
I haven't really decided here. If you think that there should be a list of unpriced assets returned because of errors we can do that. Since we're not at 1.0 yet we can modify the API structure however we think it should look.